### PR TITLE
Connection action buttons position + highlighting on invalid connections

### PR DESCRIPTION
**Pull request by wejendorp**
- Shows action buttons at the cursor position, factoring in zoom level in the zoomAndPan handler
- Uses a timeout to trigger the actionButton only after the connection has been active for a while (ala onEnter delay)
- "Rounds" the cursor position to always be exactly on the line, wherever the closest part of the line is.
- Highlights the invalid connection by darkening the dashed stroke

https://github.com/kapetacom/ui-web-plan-editor/assets/296057/d1bd4d46-4e7a-4b41-82bd-ca53a1995baa

### CODE REVIEW

**Comment by wejendorp** [comment]
npm publish

**Comment by wejendorp** [comment]
npm publish

**Comment by wejendorp** [comment]
npm publish

**Comment by wejendorp** [comment]
npm publish

File: src/planner/Planner.tsx [hunk 1]
```diff
@@ -236,6 +236,7 @@ export const Planner = (props: PlannerProps) => {
 
     const focusInfo = useFocusInfo();
     const focusModeEnabled = !!focusInfo;
+    const [panZoom, setPanZoom] = useState({ x: 0, y: 0, zoom: 1 });
```
Comment (andertun) [inline src/planner/Planner.tsx:239]: Do we need it here when we already have it in the planner context? Every component inside that react context can get it from the context

https://github.com/kapetacom/ui-web-plan-editor/blob/master/src/planner/PlannerContext.tsx#L92-L96

Comment (wejendorp) [reply src/planner/Planner.tsx:239]: No in that case we don't. I just couldn't grok it apparently. Thanks, will reuse that one instead.

**Comment by andertun** [comment]
Nice! Looking forward to this update 😊

**Pull request merged**